### ISSUE TITLE
Fix calculation in train & test loss

### ISSUE THREAD

**Issue by xiaoxi-private** [xiaoxi-s/transformer-with-shakespeare#20]
Fix the calculation in train & test loss. They should time the batch size instead of len of `batch`, which is two dimensional x, y pairs. 
```python
                training_loss += (loss.item() * len(batch))
                test_loss += (loss.item() * len(batch))
```
